apps/laboratory/src/utils/BitcoinUtil.ts

```diff
@@ -0,0 +1,141 @@
+import type { BitcoinConnector } from '@reown/appkit-adapter-bitcoin'
+import type { CaipNetwork, CaipNetworkId } from '@reown/appkit'
+import * as networks from '@reown/appkit/networks'
+import * as bitcoin from 'bitcoinjs-lib'
+
+export const BitcoinUtil = {
+  createSignPSBTParams(params: BitcoinUtil.CreateSignPSBTParams): BitcoinConnector.SignPSBTParams {
+    const network = this.getBitcoinNetwork(params.network.caipNetworkId)
+
+    const psbt = new bitcoin.Psbt({ network })
+    const payment = bitcoin.payments.p2wpkh({ address: params.senderAddress, network })
+
+    if (!payment.output) {
+      throw new Error('Invalid payment output')
+    }
+
+    const change = this.calculateChange(params.utxos, params.amount, params.feeRate)
+
+    if (change < 0) {
+      throw new Error('Insufficient funds')
+    } else if (change > 0) {
+      psbt.addOutput({
+        address: params.senderAddress,
+        value: BigInt(change)
+      })
+    }
+
+    for (const utxo of params.utxos) {
+      psbt.addInput({
+        hash: utxo.txid,
+        index: utxo.vout,
+        witnessUtxo: {
+          script: payment.output,
+          value: BigInt(utxo.value)
+        }
+      })
+    }
+
+    psbt.addOutput({
+      address: params.recipientAddress,
+      value: BigInt(params.amount)
+    })
+
+    if (params.memo) {
+      const data = Buffer.from(params.memo, 'utf8')
+      const embed = bitcoin.payments.embed({ data: [data] })
+
+      if (!embed.output) {
+        throw new Error('Invalid embed output')
+      }
+
+      psbt.addOutput({
+        script: embed.output,
+        value: BigInt(0)
+      })
+    }
+
+    return {
+      psbt: psbt.toBase64(),
+      signInputs: [],
+      broadcast: false
+    }
+  },
+
+  async getUTXOs(address: string, networkId: CaipNetworkId): Promise<BitcoinUtil.UTXO[]> {
+    const isTestnet = this.isTestnet(networkId)
+    // Make chain dynamic
+
+    const response = await fetch(
+      `https://mempool.space${isTestnet ? '/testnet' : ''}/api/address/${address}/utxo`
+    )
+
+    return await response.json()
+  },
+
+  async getFeeRate() {
+    const defaultFeeRate = 2
+    try {
+      const response = await fetch('https://mempool.space/api/v1/fees/recommended')

```

Comment:
@geekbrother any way we can have this in blockchain api ?

Reply:
yes team please let's get someone assigned to this